server/src/main/java/io/crate/expression/tablefunctions/MatchesFunction.java

```
@@ -0,0 +1,236 @@
+                final RowN row = new RowN(columns);
+                int idx = 0;
+
+                @Override
+                public boolean hasNext() {
+                    return idx < rowGroups.size();
+                }
+
+                @Override
+                public Row next() {
+                    if (!hasNext()) {
+                        throw new NoSuchElementException("no more rows");
+                    }
+                    columns[0] = rowGroups.get(idx++);
+                    return row;
+                }
+            };
+        }
+        return () -> rowIterator;
+    }
+
+    private static RegexMatcher createMatcher(String regex, @Nullable String flags) {
+
+        Function<Supplier<List<String>>, List<List<String>>> collector =
+            RegexMatcher.isGlobal(flags) ?
+                (supplier) -> {
+                    List<List<String>> result = Stream
+                        .generate(supplier)
+                        .takeWhile(Objects::nonNull)
+                        .collect(Collectors.toList());
+                    return result.isEmpty() ? null : result;
+                }
+                :
+                (supplier) -> {
+                    List<String> groups = supplier.get();
+                    return groups != null ? List.of(groups) : null;
+                };
+        return new RegexMatcher(regex, flags) {
+
+            @Override
+            public List<List<String>> match(String term) {
+                Matcher matcher = initMatcher(term);
+                return collector.apply(() -> {
+                    if (!matcher.find()) {
+                        return null;
+                    }
+                    int groupCount = matcher.groupCount();
+                    if (groupCount == 0) {
+                        try {
+                            return List.of(matcher.group());
+                        } catch (IllegalStateException e) {
+                            return null;
```

i am wondering why this exception is swallowed and the null is returned, shouldn't we propagate this failure?

---

```
matcher.group() throws IllegalStateException:
     *  If no match has yet been attempted,
     *  or if the previous match operation failed
```

Removing try/catch block to propagate the exception :)